dot/parachain/network-bridge/receiver.go

```diff
@@ -42,56 +227,411 @@ func (nbr *NetworkBridgeReceiver) Name() parachaintypes.SubSystemName {
 
 func (nbr *NetworkBridgeReceiver) ProcessActiveLeavesUpdateSignal(
 	signal parachaintypes.ActiveLeavesUpdateSignal) error {
+
+	// TODO update cpvs.activeLeaves by adding new active leaves and removing deactivated ones
+
+	// TODO: get the value for majorSyncing for syncing package
+	// majorSyncing means you are 5 blocks behind the tip of the chain and thus more aggressively
+	// download blocks etc to reach the tip of the chain faster.
+	var majorSyncing bool
+
+	nbr.liveHeads = append(nbr.liveHeads, parachaintypes.ActivatedLeaf{
+		Hash:   signal.Activated.Hash,
+		Number: signal.Activated.Number,
+	})
+
+	newLiveHeads := []parachaintypes.ActivatedLeaf{}
+
+	for _, head := range nbr.liveHeads {
+		if slices.Contains(signal.Deactivated, head.Hash) {
+			newLiveHeads = append(newLiveHeads, head)
+		}
+	}
+
+	sort.Sort(SortableActivatedLeaves(newLiveHeads))
+	nbr.liveHeads = newLiveHeads
+
+	if !majorSyncing {
+		// update our view
+		err := nbr.updateOurView()
+		if err != nil {
+			return fmt.Errorf("updating our view: %w", err)
+		}
+	}
+	return nil
+}
+
+type SortableActivatedLeaves []parachaintypes.ActivatedLeaf
+
+func (s SortableActivatedLeaves) Len() int {
+	return len(s)
+}
+
+func (s SortableActivatedLeaves) Less(i, j int) bool {
+	return s[i].Number > s[j].Number
+}
+
+func (s SortableActivatedLeaves) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (nbr *NetworkBridgeReceiver) updateOurView() error {
+	headHashes := []common.Hash{}
+	for _, head := range nbr.liveHeads {
+		headHashes = append(headHashes, head.Hash)
+	}
+	newView := View{
+		heads:           headHashes,
+		finalizedNumber: nbr.finalizedNumber,
+	}
+
+	if nbr.localView == nil {
+		*nbr.localView = newView
+		return nil
+	}
+
+	if nbr.localView.checkHeadsEqual(newView) {
+		// nothing to update
+		return nil
+	}
+
+	*nbr.localView = newView
+
+	// TODO: send ViewUpdate to all the collation peers and validation peers (v1, v2, v3)
+	// https://github.com/paritytech/polkadot-sdk/blob/aa68ea58f389c2aa4eefab4bf7bc7b787dd56580/polkadot/node/network/bridge/src/rx/mod.rs#L969-L1013
+
+	// TODO: Create our view and send collation events to all subsystems about our view change
+	// Just create the network bridge and do both of these tasks as part of those. That's the only way it makes sense.
+
```

Comment:
Planning to handle in this PR?
Or do we have an issue for this?

Reply:
tagged #4156 in the codebase here.